### PR TITLE
Add methods to remove hooks

### PR DESCRIPTION
**Pull request by KusStar**
## Changes

- null guard for sqliteExecute params: `results` and `metadatas`, I accidentally call `SELECT * FROM TABLE` with `executeBatch` it crash, cause null pointer error.

- add methods to remove hooks, so we can implement features like add hook when enter or remove hook when exit.

### CODE REVIEW

**Comment by ospfranco** [comment]
Increasing the surface of the JS API is not something I prefer. I think it is better to stick to the semantics of the original functions. Pass a null value to remove the hook:

```ts
// remove the listener
updateHook(null)
```

**Comment by ospfranco** [comment]
Thanks! But I also meant re-using the JSI functions

**Comment by KusStar** [comment]
Alright, I will make another commit.

File: cpp/bridge.cpp [hunk 1]
```diff
@@ -367,27 +367,30 @@ namespace opsqlite {
                             }
                             i++;
                         }
-                        results->push_back(row);
+                        if (results != nullptr) {
```
Comment (ospfranco) [inline cpp/bridge.cpp:370]: This is not related to the hooks changes right? Why change it? the `results` vector and the `metadatas` should always be there if I recall correctly

Comment (ospfranco) [reply cpp/bridge.cpp:370]: If you really want to check for the null pointer then I suggest you move to the top of the function and throw an exception there. I prefer the early return/fail pattern than to silently do nothing.

File: example/src/tests/hooks.ts [hunk 2]
```diff
@@ -65,6 +65,33 @@ export function registerHooksTests() {
...
+      );
+
+      db.updateHook(null)
+
+      db.execute(
+        'INSERT INTO "User" (id, name, age, networth) VALUES(?, ?, ?, ?)',
+        [id + 1, name, age, networth],
+      );
+
+      await sleep(100)
```
Comment (ospfranco) [inline example/src/tests/hooks.ts:90]: Can you try with a await new Promise(resolve => resolve()) or something similar? If not this adds 100 ms latency to the tests and we only care the callback is not trigged on the next tick

**Review by ospfranco — COMMENTED** [review]
Thanks a lot! Left some comments

File: cpp/bridge.cpp (hunk 1 above)
Comment (KusStar) [reply cpp/bridge.cpp:370]: https://github.com/OP-Engineering/op-sqlite/blob/1c0cef742d117cd67aae6a80760cbf441d074901/cpp/sqlbatchexecutor.cpp#L63-L64

In `sqlbatchexecutor` function, it call sqliteExecute with nullptr results and metadatas, I think it is intentional ignoring them.

File: example/src/tests/hooks.ts (hunk 2 above)
Comment (KusStar) [reply example/src/tests/hooks.ts:90]: Thus remove hook methods have no callbacks, cannot be resolved after remove the hook, if with new Promise(...) would be too hacky

```ts
	it('remove rollback hook', async () => {
      let promiseResolve: any;
      let promise = new Promise(resolve => {
        promiseResolve = resolve;
      })
      const hookRes: any[] = [];
      db.rollbackHook(() => {
        hookRes.push('rollback');
        promiseResolve?.();

        promise = new Promise(resolve => {
          promiseResolve = resolve;
        })
      });

      try {
        await db.transaction(async tx => {
          throw new Error('Blah');
        });
      } catch (e) {
        // intentionally left blank
      }
      await promise

      db.rollbackHook(null)
      promiseResolve?.();

      try {
        await db.transaction(async tx => {
          throw new Error('Blah');
        });
      } catch (e) {
        // intentionally left blank
      }
      await promise

      expect(hookRes.length).to.equal(1)
    });
```

Comment (ospfranco) [reply example/src/tests/hooks.ts:90]: no, that's not what I mean. I mean awaiting an empty promise to make sure the callback queue has ticked

```ts
await Promise.resolve()
```

Comment (KusStar) [reply example/src/tests/hooks.ts:90]: How about `await sleep(0)`? 
I try with await Promise.resolve() the result was not correct, but with `await sleep(0)` is right. Maybe the callback was ticked in next loop.

```js
db.execute(
  'INSERT INTO "User" (id, name, age, networth) VALUES(?, ?, ?, ?)',
  [id, name, age, networth],
);

db.updateHook(null)

db.execute(
  'INSERT INTO "User" (id, name, age, networth) VALUES(?, ?, ?, ?)',
  [id + 1, name, age, networth],
);

// if run with `await Promise.resolve()`, the expect hookRes.length = 0, is wrong, need to be 1.
await sleep(0)

expect(hookRes.length).to.equal(1)
```

File: cpp/bridge.cpp (hunk 1 above)
Comment (ospfranco) [reply cpp/bridge.cpp:370]: Hmmm I see... maybe we should change it, batch executor instead of using the normal execute, use the executeLiteral one

File: example/src/tests/hooks.ts (hunk 2 above)
Comment (ospfranco) [reply example/src/tests/hooks.ts:90]: Sounds good

**Review by ospfranco — APPROVED** [review]
